server/game/cards/06-CotE/UnveiledDestiny.js

```diff
@@ -0,0 +1,16 @@
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class UnveiledDestiny extends DrawCard {
+    setupCardAbilities() {
+        this.persistentEffect({
+            condition: context => context.player.role && context.source.parent.isAttacking(),
+            match: ring => ring.contested,
+            effect: AbilityDsl.effects.addElement(context => context.player.role.getElement())
```

Comment:
The `calculate` function for dynamic effects gets passed the target of the effect first (the contested ring in this case), so you need `(ring, context) => context.player.role.getElement()` here